e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,331 @@
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.SkipMultiRepo)
+
+	nt.T.Log("Add a deployment to a directory")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add a deployment to a directory")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Ensure that the system created a sync for the deployment")
+	_, err := nt.Kubectl("get", "sync", "deployment.apps")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Force-delete the sync from the cluster")
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("delete", "sync", "deployment.apps")
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Ensure that the system re-created the force-deleted sync")
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("get", "sync", "deployment.apps")
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestDeletingClusterConfigRecoverable(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.SkipMultiRepo)
+
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add dir namespace")
+	nt.WaitForRepoSyncs()
+
+	_, err := nt.Kubectl("get", "ns", "dir")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Forcefully delete clusterconfigs and verify recovery")
+	_, err = nt.Kubectl("delete", "clusterconfig", "--all")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.RootRepos[configsync.RootSyncName].Copy("../../examples/acme/cluster/admin-clusterrole.yaml", "acme/cluster/admin-clusterrole.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add admin clusterrole")
+	nt.WaitForRepoSyncs()
+
+	_, err = nt.Kubectl("get", "clusterconfig", "config-management-cluster-config")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestNamespaceGarbageCollection(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.SkipMultiRepo)
+
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/accounting-namespace.yaml", yamlDir), "acme/namespaces/accounting/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add accounting namespace")
+	nt.WaitForRepoSyncs()
+	_, err := nomostest.Retry(60*time.Second, func() error {
+		_, err := nt.Kubectl("get", "ns", "accounting")
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/accounting/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove accounting namespace")
+	nt.WaitForRepoSyncs()
+
+	_, err = nt.Kubectl("get", "ns", "accounting")
+	if err == nil {
```

Comment:
These `err == nil` checks would be more precise as `apierrors.IsNotFound`. Otherwise it will fail silently for other unexpected error conditions

Reply:
replaced validation to use nt.ValidateNotFound